test/mapping.test.js

```diff
@@ -18,19 +18,36 @@ describe('Mapping', function () {
 
   describe('ircToSlack', function () {
     before(function () {
-      mapping.link('slackname', 'ircusername');
+      mapping.link('coolname', 'rob');
     });
 
     after(function () {
-      mapping.unlink('slackname', 'ircusername');
-    })
+      mapping.unlink('coolname', 'rob');
+    });
 
     it('does not replace names in URLS', function () {
-      var message = 'Check out http://github.com/ircusername/coolrepo';
+      var message = 'Check out http://github.com/rob/coolrepo';
       var newMessage = mapping.ircToSlack(message);
       expect(newMessage).to.equal(message);
     });
 
-  });
+    it('does not replace names in the middle of other words', function () {
+      var message = 'Worry, not it is not a problem';
+      var newMessage = mapping.ircToSlack(message);
+      expect(newMessage).to.equal(message);
+    });
+
+    it('does the replacement with a non-letter character afterwards', function () {
+      var message = 'rob: what is the plan?';
+      var newMessage = mapping.ircToSlack(message);
+      expect(newMessage).to.equal('<@coolname>: what is the plan?');
+    });
+
+    it('does the replacement when the message is just a one word question', function () {
+      var message = 'rob?';
+      var newMessage = mapping.ircToSlack(message);
+      expect(newMessage).to.equal('<@coolname>?');
+    });
```

Comment:
let's add a test for `@rob` too, to make sure that doesn't get lost as we iterate on this.